pyanaconda/modules/storage/partitioning/automatic/automatic_partitioning.py

```diff
@@ -122,37 +123,59 @@ def _get_luks_format_args(storage, request):
         }
 
     @staticmethod
-    def _get_partitioning(storage, scheme, excluded_mount_points=()):
+    def _get_partitioning(storage, scheme, request: PartitioningRequest):
         """Get the partitioning requests for autopart.
 
         :param storage: blivet.Blivet instance
         :param scheme: a type of the partitioning scheme
-        :param excluded_mount_points: a list of mount points to exclude
+        :param request: partitioning parameters
         :return: a list of full partitioning specs
         """
-        requests = []
+        specs = []
+        will_swap_be_created = False
 
-        for request in get_default_partitioning():
+        for spec in get_default_partitioning():
```

Comment:
Feel free to split the loop. The effectivity of the code is usually not a concern for the installer.

1. Filter out unwanted specs.
2. Add a spec for swap if hibernation is requested and there is no swap.
3. Configure the specs.

Then the specs are configured the same way no matter where they come from.